tests/tasks/test_refresh_schemas.py

```diff
@@ -188,3 +189,90 @@ def test_refresh_schema_update_column(self):
+                    'name': NEW_COLUMN_NAME,
+                    'type': self.COLUMN_TYPE,
+                }]
+            })
+
+        self.patched_get_schema.return_value = tables
+        self.factory.data_source.query_runner.configuration['samples'] = True
+
+        refresh_schema(self.factory.data_source.id)
+        get_all_table_samples(self.factory.data_source.id)
+
+        # There's a total of 105 tables
+        table_metadata = TableMetadata.query.all()
+        self.assertEqual(len(table_metadata), NUM_TABLES)
+
+        # 50 tables are processed on the first call
+        table_metadata = TableMetadata.query.filter(
+            TableMetadata.latest_sampling.is_(None)
+        ).all()
+        self.assertEqual(len(table_metadata), 55)
+
+        # 50 more tables are processed on the second call
+        get_all_table_samples(self.factory.data_source.id)
+        table_metadata = TableMetadata.query.filter(
+            TableMetadata.latest_sampling.is_(None)
+        ).all()
+        self.assertEqual(len(table_metadata), 5)
+
+        # All tables are processed by the third call
+        get_all_table_samples(self.factory.data_source.id)
+        table_metadata = TableMetadata.query.filter(
+            TableMetadata.latest_sampling.is_(None)
+        ).all()
+        self.assertEqual(len(table_metadata), 0)
+
+    def test_refresh_schema_get_all_table_samples_refreshes(self):
+        NEW_COLUMN_NAME = 'new_column'
+        NUM_TABLES = 5
+        TIME_BEFORE_UPDATE = utils.utcnow()
+        tables = []
+
+        for i in range(NUM_TABLES):
+            tables.append({
+                'name': 'table{}'.format(i),
+                'columns': [NEW_COLUMN_NAME],
+                'metadata': [{
+                    'name': NEW_COLUMN_NAME,
+                    'type': self.COLUMN_TYPE,
+                }]
+            })
+
+        self.patched_get_schema.return_value = tables
+        self.factory.data_source.query_runner.configuration['samples'] = True
+
+        refresh_schema(self.factory.data_source.id)
+        get_all_table_samples(self.factory.data_source.id)
+
+        # There's a total of 5 processed tables
+        table_metadata = TableMetadata.query.filter(
+            TableMetadata.latest_sampling.isnot(None)
+        )
+        table_metadata_list = table_metadata.all()
+        self.assertEqual(len(table_metadata_list), NUM_TABLES)
+        self.assertTrue(table_metadata_list[0].latest_sampling > TIME_BEFORE_UPDATE)
+
+        table_metadata.update({
+            'latest_sampling': utils.utcnow() - datetime.timedelta(days=30)
+        })
+        models.db.session.commit()
+
+        TIME_BEFORE_UPDATE = utils.utcnow()
+        get_all_table_samples(self.factory.data_source.id)
+        table_metadata_list = TableMetadata.query.filter(
+            TableMetadata.latest_sampling.isnot(None)
+        ).all()
+        self.assertTrue(table_metadata_list[0].latest_sampling > TIME_BEFORE_UPDATE)
```

Comment:
It suffices to use `first()` here instead of loading all TableMetadata as a list.